oracle/datadog_checks/oracle/oracle.py

```diff
@@ -233,16 +235,26 @@ def _jdbc_connect(self):
         try:
             with jdbc_lock:
                 if jpype.isJVMStarted() and not jpype.isThreadAttachedToJVM():
+                    self.log.debug("JVM started but thread not attached to JVM.")
                     jpype.attachThreadToJVM()
                     jpype.java.lang.Thread.currentThread().setContextClassLoader(
                         jpype.java.lang.ClassLoader.getSystemClassLoader()
                     )
                 connection = jdb.connect(
                     self.ORACLE_DRIVER_CLASS, connect_string, jdbc_connect_properties, self._jdbc_driver
                 )
+                if jpype.isJVMStarted() and jpype.isThreadAttachedToJVM():
+                    jpype.detachThreadFromJVM()
+                    self.log.debug("Detaching thread from JVM after connection")
```

Comment:
Any chance we can test this?

Reply:
@yzhan289 friendly bump on this one 🙂

Reply:
Unfortunately I don't know if there's an easy way we can test this since the logic happens in `jpype` and we would need to check the JVM.